IQKeyboardManager/Categories/IQUIView+Hierarchy.m

```diff
@@ -89,35 +89,54 @@ -(UIViewController*)viewController
     return nil;
 }
 
--(UIViewController *)topMostController
+- (UIViewController *)topMostController
 {
-    NSMutableArray *controllersHierarchy = [[NSMutableArray alloc] init];
-    
-    UIViewController *topController = self.window.rootViewController;
-    
-    if (topController)
+    UIViewController *result = nil;
+    UIWindow *window = [[UIApplication sharedApplication] keyWindow];
+    if (window.windowLevel != UIWindowLevelNormal)
```

Comment:
What if user created his own window hierarchy? You are only trying to get root view controller.